File: src/main/java/com/gusrylmubarok/reddit/backend/model/Post.java
package com.gusrylmubarok.reddit.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.lang.Nullable;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.time.Instant;
import java.util.List;

@Entity
@Table(name = "post")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "post_name")
    @NotBlank(message = "post name cannot be empty or null!")
    private String postName;

    @Column(name = "url")
    @Nullable
    private String url;

    @Column(name = "description")
    @Nullable
    @Lob
    private String description;

    @Column(name = "vote_count")
    private int voteCount = 0;

    @Column(name = "created_date")
    private Instant createdDate;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "subreddit_id", referencedColumnName = "id")
    private Subreddit subreddit;

//    @OneToMany(mappedBy = "post")
//    private List<Comment> comments;
//
//    @OneToMany(mappedBy = "post")
//    private List<Vote> votes;

}

File: src/main/kotlin/com/reddot/data/entity/Post.kt
package com.reddot.data.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "post")
data class Post(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,

    @Column(name = "post_name")
    var postName: String,

    @Column(name = "url")
    var url: String,

    var description: String,

    @Column(name = "vote_count")
    var voteCount: Int,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userId", referencedColumnName = "id")
    var user: User,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subredditId", referencedColumnName = "id")
    var subreddit: Subreddit,

    @Column(name = "created_at")
    var createdAt: Date,

    @Column(name = "updated_at")
    var updatedAt: Date?
)
